.ci/validation.sh

```diff
@@ -167,6 +167,11 @@ test-tr)
     -DargLine='-Duser.language=tr -Duser.country=TR -Xms1024m -Xmx2048m'
   ;;
 
+test-ru)
+  mvn -e --no-transfer-progress clean integration-test failsafe:verify \
+    -DargLine='-Duser.language=ru -Duser.country=RU -Xms1024m -Xmx2048m'
```

Comment:
Can we not rework these so instead of 20 "cmd test-XX", can we do "cmd test-language XX YY"?

Reply:
@strkkk , do you know why CI for Run is green?

Reply:
No idea

Reply:
We need find way to fail CI in other cases we never resolve this issue

Reply:
Running `integration-test failsafe:verify -DargLine='-Duser.language=ru -Duser.country=RU'` on my local,

I am getting output saying my locale is still english.

````
[INFO] Running com.puppycrawl.tools.checkstyle.api.ViolationTest
1: en_US
2: en
[WARNING] Tests run: 22, Failures: 0, Errors: 0, Skipped: 1, Time elapsed: 0.1 s - in com.puppycrawl.tools.checkstyle.api.ViolationTest
````

````
diff --git a/src/test/java/com/puppycrawl/tools/checkstyle/api/ViolationTest.java b/src/test/java/com/puppycrawl/tools/checkstyle/api/ViolationTest.java
index efccf13..01cdacb 100644
--- a/src/test/java/com/puppycrawl/tools/checkstyle/api/ViolationTest.java
+++ b/src/test/java/com/puppycrawl/tools/checkstyle/api/ViolationTest.java
@@ -84,10 +84,13 @@
      */
     @Test
     public void testLocaleIsSupported() {
+        System.out.println("1: " + DEFAULT_LOCALE);
         final String language = DEFAULT_LOCALE.getLanguage();
+        System.out.println("2: " + language);
         assumeFalse(language.isEmpty() || Locale.ENGLISH.getLanguage().equals(language),
                 "Custom locale not set");
         final Violation violation = createSampleViolation();
+        System.out.println("3: " + violation.getViolation());
         assertWithMessage("Unsupported language: %s", DEFAULT_LOCALE)
                 .that(violation.getViolation())
                 .isNotEqualTo("Empty statement.");
````

Reply:
It means that plugin does not use value of argLine, can you change plugin in pom.xml to get locale to be set.

Reply:
Russian run in CI says items were skipped.
> [WARNING] Tests run: 22, Failures: 0, Errors: 0, Skipped: 2, Time elapsed: 0.227 s - in com.puppycrawl.tools.checkstyle.api.ViolationTest

3 tests have the assumes in this class, so it doesn't say if we can connect this.

Is there a way to make assumes print out if they are causing the test to skip?

Reply:
Change in pom is producing errors.
````
diff --git a/pom.xml b/pom.xml
index 34aee2a..4ec154a 100644
--- a/pom.xml
+++ b/pom.xml
@@ -197,7 +197,7 @@
   <properties>
     <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
     <project.reporting.outputEncoding>UTF-8</project.reporting.outputEncoding>
-    <surefire.options />
+    <surefire.options>-Duser.language=ru -Duser.country=RU</surefire.options>
     <projectVersion>${project.version}</projectVersion>
     <antlr4.version>4.9.3</antlr4.version>
     <maven.site.plugin.version>3.11.0</maven.site.plugin.version>
````

````
[INFO] Running com.puppycrawl.tools.checkstyle.api.ViolationTest
1: ru_RU
2: ru
3: Empty statement.
[ERROR] Tests run: 22, Failures: 1, Errors: 0, Skipped: 0, Time elapsed: 0.147 s <<< FAILURE! - in com.puppycrawl.tools.checkstyle.api.ViolationTest
[ERROR] testLocaleIsSupported  Time elapsed: 0.062 s  <<< FAILURE!
com.google.common.truth.AssertionErrorWithFacts: 
Unsupported language: ru_RU
expected not to be: Empty statement.
	at com.puppycrawl.tools.checkstyle.api.ViolationTest.testLocaleIsSupported(ViolationTest.java:96)
````

Reply:
Full failure list:
````
[ERROR] Failures: 
[ERROR]   ViolationTest.testLocaleIsSupported:96 Unsupported language: ru_RU
expected not to be: Empty statement.
[ERROR] Errors: 
[ERROR]   DefaultLoggerTest.testLocaleIsSupported:254 » NoSuchMethod com.puppycrawl.tool...


[ERROR] testLocaleIsSupported  Time elapsed: 0 s  <<< ERROR!
java.lang.NoSuchMethodException: com.puppycrawl.tools.checkstyle.DefaultLogger$LocalizedMessage.<init>(java.lang.String, [Ljava.lang.String;)
	at com.puppycrawl.tools.checkstyle.DefaultLoggerTest.testLocaleIsSupported(DefaultLoggerTest.java:254)
````

Reply:
Seems strange that `surefire.options` is pulled into `argLine` but setting `argLine` directly doesn't work...

Reply:
https://dev.azure.com/romanivanovjr/romanivanovjr/_build/results?buildId=8630&view=logs&j=3b0b6035-d26e-5411-a529-6e236284dc45&t=fdcfc88d-226d-5705-159e-79095a68acd9
>  (s) argLine = -Dfile.encoding=UTF-8 

Seems like argLine isn't being replaced.

Reply:
CI is now reproducing my local.

Reply:
>ViolationTest.testLocaleIsSupported:96 Unsupported language: ru_RU expected not to be: Empty statement.

ok, as we fix it, this should not be a failure.
Lets keep RU (or we can select some other very unlikely used language) execution as prove that tests are passing on it.

Reply:
WE can take 1st from https://www.oracle.com/java/technologies/javase/java8locales.html


Language (ISO 639) | Country (ISO 3166) | Variant | Script (ISO 15924) | Language Tag | Note
-- | -- | -- | -- | -- | --
Albanian (sq) | Albania (AL) |   | (Latn) | sq-AL |  